agent-any/src/AnyAgentComplex.cc

```diff
@@ -511,6 +517,13 @@ AnyAgent::parseData (char const *&line, const YCPValue & syntax, bool optional)
 		tupleName.push ("");
 		tupleValue.push (YCPNull ());
 		YCPValue tv = parseTuple (line, term->args (), optional);
+        if (tv.isNull())
+        {
+            if (optional)
+                y2milestone("Optional Tuple not found at line: '%s'", line);
+            else
+                y2error("Tuple not found at line: '%s'", line);
```

Comment:
Um, I'm not sure, the line might not contain a new line (the last line), the newline is processed by the agent definition so in some case you actually might know whether there is a new line or not.

Yes, it makes the logging a bit less readable but I'd rather prefer full info in the log.